.github/workflows/CI.yml

```diff
@@ -86,10 +86,13 @@ jobs:
         if: runner.os == 'macOS'
         # restrict number of openMP threads on macOS due to oversubscription
         run: echo "OMP_NUM_THREADS=1" >> $GITHUB_ENV
-      - name: "Use multiple processes for self-hosted runners"
+      - name: "Use multiple processes for self-hosted macOS runners"
         if: runner.os == 'macOS' && runner.environment == 'self-hosted'
         # runner.environment is supposed to be a valid property: https://github.com/orgs/community/discussions/48359#discussioncomment-9059557
         run: echo "NUMPROCS=5" >> $GITHUB_ENV
+      - name: "Use multiple processes for self-hosted Linux runners"
+        if: runner.os == 'Linux' && runner.environment == 'self-hosted'
+        run: echo "NUMPROCS=6" >> $GITHUB_ENV
```

Comment:
Can't this be made part of the runner config instead?

Reply:
There's no unified way of doing this (that I know of), one would have to set the environment variable for each runner (and then if/when we update this, that will have to be changed manually for each runner).

Reply:
Can't we use https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job to set the `NUMPROC` environment variable *on the runners*? IMHO that's really were this setting belongs (and then we can decide on the runners how many cores we want to assign)